src/ORM/Table.php

```diff
@@ -2365,6 +2385,16 @@ public function buildRules(RulesChecker $rules)
     }
 
     /**
+     * Gets a SaveOptionsBuilder instance.
+     *
+     * @return \Cake\ORM\SaveOptionsBuilder
+     */
+    public function getSaveOptionsBuilder()
+    {
+        return new SaveOptionsBuilder();
```

Comment:
Yes, it's still WIP, I've been chatting with Jose a moment ago about this and just PR'ed for an easier review than looking at each of the commits in my branch.